httpx/client_test.go

```diff
@@ -43,6 +44,41 @@ func (c *FailingClient) Do(*http.Request) (*http.Response, error) {
 	}, nil
 }
 
+type MetricsInstrument struct {
+	metrics.Metrics
+	timerMock metrics.Timer
+}
+
+func (mi *MetricsInstrument) Timer(name string, tags ...metrics.Tag) metrics.Timer {
+	return mi.timerMock
+}
+
+type TimerMock struct {
+	metrics.Timer
+	TagsMap map[string]interface{}
```

Comment:
I don't see any mutex around, so this code is not concurrent-safe. If you ever use this in a concurrent test it will fail.